# 텐서플로 2.0 시작하기: 초보자용

In [1]:
import tensorflow as tf

[MNIST 데이터셋](http://yann.lecun.com/exdb/mnist/)을 로드하여 준비합니다. 샘플 값을 정수에서 부동소수로 변환합니다:

In [10]:
import numpy as np
mnist = tf.keras.datasets.mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

print(type(x_train))
print(x_train.shape)
print(x_test.shape)
print(np.min(x_train))
print(np.max(x_train))

<class 'numpy.ndarray'>
(60000, 28, 28)
(10000, 28, 28)
0.0
1.0


층을 차례대로 쌓아 `tf.keras.Sequential` 모델을 만듭니다. 훈련에 사용할 옵티마이저(optimizer)와 손실 함수를 선택합니다:

In [13]:
model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28)),   # (60000,28,28) => (60000,784)
  tf.keras.layers.Dense(128, activation='relu'),   # (60000,784)(784,128)+(128,) => (60000,128) , 100480
  tf.keras.layers.Dropout(0.2),                    
  tf.keras.layers.Dense(10, activation='softmax')  # (60000,128)(128,10)+(10,) => (60000,10) , 1290
])

model.summary()

model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

Model: "sequential_2"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
flatten_2 (Flatten)          (None, 784)               0         
_________________________________________________________________
dense_4 (Dense)              (None, 128)               100480    
_________________________________________________________________
dropout_2 (Dropout)          (None, 128)               0         
_________________________________________________________________
dense_5 (Dense)              (None, 10)                1290      
Total params: 101,770
Trainable params: 101,770
Non-trainable params: 0
_________________________________________________________________


모델을 훈련하고 평가합니다:

In [14]:
model.fit(x_train, y_train, epochs=5)

model.evaluate(x_test,  y_test, verbose=2)

Epoch 1/5
1875/1875 [==============================] - 8s 4ms/step - loss: 0.4715 - accuracy: 0.8617
Epoch 2/5
1875/1875 [==============================] - 7s 4ms/step - loss: 0.1510 - accuracy: 0.9564
Epoch 3/5
1875/1875 [==============================] - 7s 4ms/step - loss: 0.1046 - accuracy: 0.9678
Epoch 4/5
1875/1875 [==============================] - 7s 4ms/step - loss: 0.0876 - accuracy: 0.9729
Epoch 5/5
1875/1875 [==============================] - 7s 4ms/step - loss: 0.0716 - accuracy: 0.9773
313/313 - 0s - loss: 0.0753 - accuracy: 0.9775


[0.07531937956809998, 0.9775000214576721]

훈련된 이미지 분류기는 이 데이터셋에서 약 98%의 정확도를 달성합니다. 더 자세한 내용은 [TensorFlow 튜토리얼](https://www.tensorflow.org/tutorials/)을 참고하세요.